examples/upgradable-contract/README.md

```diff
@@ -0,0 +1,43 @@
+# Proxy Smart Contract
+
+The proxy smart contract delegates any call that does not match a
+selector of itself to another, specified contract.
+
+The instantiator of the proxy contract on a blockchain can change
+the code to which calls are forwarded.
+
+This allows building upgradable contracts following the proxy pattern.
+
+In order to test it out you need to do the following:
+
+1. Build a contract containing some logic, e.g. our upgradeable flipper example:
+   ```
+   cargo +nightly contract build --manifest-path=examples/upgradable-contract/upgradeable_flipper/Cargo.toml
+   ```
+   You will receive the respective `upgradeable_flipper.contract` bundle in the `examples/upgradable-contract/upgradeable_flipper/target/ink/` folder.
+
+   In order to perform migrations and have proxy working for contracts with different storage
+   layouts, we use the [`Upgradeable`](upgradeable_flipper/upgradeable.rs) type wrapper, which ensures
```

Comment:
We have the term "upgradeable" and "upgradable" in this PR. @ascjones :gb: Wdyt?

Reply:
I will rename everything to `upgradeable`

Reply:
We've had this debate over on the Substrate side, and American spelling should be used 🇺🇸